adapters/clients/client.go

```diff
@@ -28,7 +28,9 @@ type retryClient struct {
 
 func (c *retryClient) doWithCustomMarshaller(timeout time.Duration,
 	req *http.Request, data []byte, decode func([]byte) error, success func(code int) bool,
+	numRetries int,
 ) (err error) {
+	// TODO when this context times out and is cancelled, does it mean we won't retry up above via Pull?
```

Comment:
**note:** it means it will cancel any child and their children associated to this ctx but not the parents 

w.r.t https://pkg.go.dev/context
>The [WithCancel](https://pkg.go.dev/context#WithCancel), [WithDeadline](https://pkg.go.dev/context#WithDeadline), and [WithTimeout](https://pkg.go.dev/context#WithTimeout) functions take a Context (the parent) and return a derived Context (the child) and a [CancelFunc](https://pkg.go.dev/context#CancelFunc). Calling the CancelFunc cancels the child and its children, removes the parent's reference to the child, and stops any associated timers. Failing to call the CancelFunc leaks the child and its children until the parent is canceled or the timer fires. The go vet tool checks that CancelFuncs are used on all control-flow paths.